experiment/simulation/js/main.js

```diff
@@ -1 +1,559 @@
-//Your JavaScript goes in here
+"use strict";
+// declaring the global constants
+const MAXX = 1195;
+const MINX = 60;
+const MINY = 20;
+const MAXY = 490;
+const width = 1200;
+const height = 600;
+const textHeight = 550;
+const scale = window.devicePixelRatio;
+const gridColor = "yellow";
+const lineColor = "white";
+const highlightColor = "red";
+const INSIDE = 0; // 0000
+const LEFT = 1; // 0001
+const RIGHT = 2; // 0010
+const BOTTOM = 4; // 0100
+const TOP = 8; // 1000
+// global variables
+let times_next_called = 0;
+const canvas = document.getElementById("canvas");
+// marks the 4 borders of the rectangular grid
+let left, up, right, down;
+// mark the line coordinates
+let point1 = [],
+  point2 = [];
+let first_points = [];
+let second_points = [];
+let MESSAGE = "";
```

Comment:
Please us upper case naming strictly for constants. regular variables should follow camelCase
Also if you move the declaration of EMPTY before that of message, you can use EMPTY to assign the initial value to message